### PR TITLE
document CComponent

### PR DESCRIPTION
**Pull request by edg-l**
I think this will make it easier for future people looking into this class.

## Checklist

- [ ] Tested the change ingame
- [ ] Provided screenshots if it is a visual change
- [ ] Tested in combination with possibly related configuration options
- [ ] Written a unit test if it works standalone, system.c especially
- [ ] Considered possible null pointers and out of bounds array indexing
- [ ] Changed no physics that affect existing maps
- [ ] Tested the change with [ASan+UBSan or valgrind's memcheck](https://github.com/ddnet/ddnet/#using-addresssanitizer--undefinedbehavioursanitizer-or-valgrinds-memcheck) (optional)

### CODE REVIEW

File: src/game/client/component.h [hunk 1]
```diff
@@ -192,14 +192,14 @@ class CComponent
 	 */
 	virtual void OnMessage(int Msg, void *pRawMsg) {}
 	/**
-	 * Called on mouse movement
+	 * Called on mouse movement.
 	 *
 	 * @param x The x coordinates of the mouse.
```
Comment (Jupeyy) [inline src/game/client/component.h:197]: rn this is relative mouse movement

File: src/game/client/component.h [hunk 2]
```diff
@@ -18,61 +18,190 @@
 
 class CGameClient;
 
+/**
+* This class is inherited by all the client components.
+*
+* These components can implement the virtual methods such as OnInit(), OnMessage(int Msg, void *pRawMsg) to provide their functionality.
```
Comment (Jupeyy) [inline src/game/client/component.h:24]: guess could use `@see` here too, else looks good to me

Comment (edg-l) [reply src/game/client/component.h:24]: they are automatically linked btw

Comment (Jupeyy) [reply src/game/client/component.h:24]: ok

**Pull request merged**